public/app/features/explore/ShortLinkButtonMenu.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+
+import { IconName, urlUtil } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { ToolbarButton, Dropdown, Menu, Stack, ToolbarButtonRow } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import store from 'app/core/store';
+import { createAndCopyShortLink } from 'app/core/utils/shortLinks';
+import { useSelector } from 'app/types';
+
+import { getUrlStateFromPaneState } from './hooks/useStateSync';
+import { selectPanes } from './state/selectors';
+
+export function ShortLinkButtonMenu() {
+  const panes = useSelector(selectPanes);
+  const [isOpen, setIsOpen] = useState(false);
+  const onCopyShortLink = (persistType?: string, url?: string) => {
+    if (persistType !== undefined) {
+      store.set('grafana.explore.shortLinkShareType', persistType);
```

Comment:
@diegoadams Just to double check. Are we sure using icon is a good idea and it has enough visibility? Clicking on that button is probably something that people may not think about a lot so they may not notice it changed.

@gelicia Tooltip is incorrect after toggling

<img width="535" alt="Screenshot 2023-11-06 at 11 24 50" src="https://github.com/grafana/grafana/assets/745532/bccbd9bb-b67d-4e1a-ab9d-b5af8c7999ab">

Reply:
> @diegoadams Just to double check. Are we sure using icon is a good idea and it has enough visibility? Clicking on that button is probably something that people may not think about a lot so they may not notice it changed.

@ifrost , the default will still be the "copy shortened link" icon, but while the user is in a "session" the behaviour will persist as currently. This is the same pattern for the Run Query button.  Its a slow way to introduce the user to this feature. 

When the user hits refresh it should default back to the "copy shortened link" and icon. @gelicia, sorry I wasn't clear on this. 

<img width="201" alt="Screenshot 2023-11-08 at 22 26 02" src="https://github.com/grafana/grafana/assets/95694434/518fe5c4-0373-45fd-b56d-02e4c9251352">